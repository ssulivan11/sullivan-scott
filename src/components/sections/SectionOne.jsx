import React from 'react'
import PropTypes from 'prop-types'
import style from './sections.scss'

const SectionOne = ({ mainHeadingText, subHeadingText, windowSize, isActive }) => {
  const scrollDown = () => {
    window.scroll({
      top: windowSize.height,
      left: 0,
      behavior: 'smooth',
    })
  }

  return (
    <div style={{ height: windowSize.height }}>
      <div
        className={`${style['section-one']} ${isActive ? style['section-one-active'] : style['section-one-inactive']}`}>
        <div
          data-test='section-one-button'
          className={style['section-one-button']}
          onKeyDown={() => scrollDown()}
          onClick={() => scrollDown()}
          role='button'
          tabIndex='0'>
          <h1 className={`${style.heading} ${style['main-heading']}`}>{mainHeadingText}</h1>
          <h2 className={`${style.heading} ${style['sub-heading']}`}>{subHeadingText}</h2>
        </div>
      </div>
      <div className={style['section-one-background']} style={{ height: windowSize.height }} />
    </div>
  )
}

SectionOne.propTypes = {
  mainHeadingText: PropTypes.string,
  subHeadingText: PropTypes.string,
  windowSize: PropTypes.shape({
    height: PropTypes.number,
    width: PropTypes.number,
  }),
  isActive: PropTypes.bool,
}

export default SectionOne
