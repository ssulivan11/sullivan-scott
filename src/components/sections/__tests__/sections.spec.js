import React from 'react'
import { mount } from 'enzyme'

import SectionOne from '../SectionOne'

const defaultTestProps = {
  mainHeadingText: 'Main Heading text',
  subHeadingText: 'Subheading text',
  windowSize: {
    height: 500,
    widtth: 0,
  },
}

const setup = (props = defaultTestProps) => {
  const component = mount(<SectionOne {...props} />)
  return {
    getInstance: () => component,
    getSectionOne: () => component.find(SectionOne),
    getSectionOneButton: () => component.find("[data-test='section-one-button']"),
  }
}

let wrapper
describe('<SectionOne />', () => {
  wrapper = setup()

  it('renders', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('clicks well', () => {
    window.scroll = jest.fn()
    wrapper.getSectionOneButton().simulate('click')
    expect(window.scroll).toHaveBeenCalled()
  })
})
