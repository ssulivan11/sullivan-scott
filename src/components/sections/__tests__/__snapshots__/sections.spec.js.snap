// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionOne /> renders 1`] = `
Object {
  "getInstance": [Function],
  "getSectionOne": [Function],
  "getSectionOneButton": [Function],
}
`;
